homeassistant/components/withings/sensor.py

```diff
@@ -363,11 +365,31 @@ async def async_setup_entry(
     measurement_coordinator: WithingsMeasurementDataUpdateCoordinator = hass.data[
         DOMAIN
     ][entry.entry_id][MEASUREMENT_COORDINATOR]
+
+    current_measurement_types = set(measurement_coordinator.data.keys())
+
     entities: list[SensorEntity] = []
     entities.extend(
-        WithingsMeasurementSensor(measurement_coordinator, attribute)
-        for attribute in MEASUREMENT_SENSORS
+        WithingsMeasurementSensor(
+            measurement_coordinator, MEASUREMENT_SENSORS[measurement_type]
+        )
+        for measurement_type in measurement_coordinator.data
+        if measurement_type in MEASUREMENT_SENSORS
     )
+
+    def measurement_listener() -> None:
```

Comment:
I'm assuming this runs in async context?  If so it should be named `_async_measurement_listener` to indicate its async safe.